recipes/sherpa/build.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# For Sherpa v2, remove specific linker flags from LDFLAGS to ensure all libraries
+# have symbols defined
+if [[ "${build_platform}" == linux-* ]]; then
+    # On Linux remove '--as-needed' flag
+    # c.f. https://linux.die.net/man/1/ld
+    export LDFLAGS="$(echo $LDFLAGS | sed 's/ -Wl,--as-needed//g')"
+    # Sherpa installs libraries into LIBDIR $PREFIX/lib/SHERPA-MC
+    export LDFLAGS="$LDFLAGS -Wl,-rpath,$PREFIX/lib/SHERPA-MC -Wl,-rpath-link,$PREFIX/lib/SHERPA-MC -L$PREFIX/lib/SHERPA-MC"
+else
+    # On macOS remove '-dead_strip_dylibs' flag
+    # c.f. https://github.com/AnacondaRecipes/intel_repack-feedstock/issues/8
+    export LDFLAGS="$(echo $LDFLAGS | sed 's/ -Wl,-dead_strip_dylibs//g')"
+    # Sherpa installs libraries into LIBDIR $PREFIX/lib/SHERPA-MC
+    export LDFLAGS="$LDFLAGS -Wl,-rpath,$PREFIX/lib/SHERPA-MC -L$PREFIX/lib/SHERPA-MC"
+fi
```

Comment:
If the linker flags `--as-needed` (Linux) and `-dead_strip_dylibs` (macOS) aren't removed, then I get an undefined symbol at runtime (c.f. https://github.com/conda-forge/staged-recipes/pull/28084#issuecomment-2452899392).

Adding the install library path of `$PREFIX/lib/SHERPA-MC` to linker flags in `LDFLAGS` doesn't seem to help with avoiding the `Failed to load library` errors.

Reply:
As wisely explained by @chrisburr:

> I bet it's prefix replacement.
> ```
> $PREFIX/lib/SHERPA-MC: cannot read file data: Is a directory
> ```
> 
> At build time conda uses a really long environment name (the whole "placeholder" stuff) so that any hardcoded strings inside binaries are very long. Then at install time it replaces `$PREFIX/....` with `$CONDA_PREFIX/....`. The install prefix (`$CONDA_PREFIX`) is shorter than the build time prefix (`$PREFIX`) and any extra characters on the end of the string are padded with null bytes. This works well for c-style strings (which cannot contain null bytes as that terminates the string by definition) however C++ strings have a known size so they contain null bytes. So when you do `$PREFIX/lib/SHERPA-MC` + `/libSherpaSM.so` you get `$CONDA_PREFIX/lib/SHERPA-MC/\x00\x00...\x00\x00/libSherpaSM.so`.
> 
> I think you could add a patch here to remove the trailing nulls:
> https://gitlab.com/sherpa-team/sherpa/-/blob/f59a714bb661f56ecb1754bfbc734e680c8feca4/ATOOLS/Org/Run_Parameter.C#L445